packages/widgetbook/lib/src/addons/frame_addon/frames/device_frame/device_setting_provider.dart

```diff
@@ -0,0 +1,17 @@
+import 'package:flutter/material.dart';
+import 'package:widgetbook/widgetbook.dart';
+
+class DeviceSettingProvider extends ValueNotifier<DeviceSetting?> {
```

Comment:
Why is this nullable?

Reply:
This is set to nullable so that I can use DeviceSetting with Noframe Addon.
```
Widget _wrapperBuilder(
  Widget child,
) {
  return ChangeNotifierProvider(
    key: const ValueKey('NoFrameAddon'),
    create: (_) => DeviceSettingProvider(null),
    child: child,
  );
}

```
I tried without it but when switching from Noframe to the other frame I lost the `DeviceSettingProvider` instance making the device addon to brake.

The solution was to make noFrame Addon extend `WidgetbookAddOn` and set `DeviceSettingProvider`

DeviceSettingProvider takes DeviceSetting which requires 
```
  required List<Device> devices,
    required Device activeDevice,
    @Default(Orientation.portrait) Orientation orientation,
```
and they are not required for a no frame